common/flatpak-run.c

```diff
@@ -578,15 +578,28 @@ flatpak_run_add_session_dbus_args (FlatpakBwrap   *app_bwrap,
 
   unrestricted = (context->sockets & FLATPAK_CONTEXT_SOCKET_SESSION_BUS) != 0;
 
-  if (dbus_address == NULL)
-    return FALSE;
+  if (dbus_address != NULL)
+    {
+      dbus_session_socket = extract_unix_path_from_dbus_address (dbus_address);
+    }
+  else
+    {
+      g_autofree char *user_runtime_dir = flatpak_get_real_xdg_runtime_dir ();
+      struct stat statbuf;
+
+      dbus_session_socket = g_build_filename (user_runtime_dir, "bus", NULL);
+
+      if (stat (dbus_session_socket, &statbuf) < 0
+          || (statbuf.st_mode & S_IFMT) != S_IFSOCK
+          || statbuf.st_uid != getuid ())
+        return FALSE;
```

Comment:
This `return` leaks `dbus_session_socket`. I suppose that the variable should be made `g_autofree`?

Reply:
Good catch! I think this was a pre-existing bug: it would previously have leaked the result of `extract_unix_path_from_dbus_address()` if that function call was successful. Please see #2877.